katportalclient/client.py

```diff
@@ -1421,6 +1345,249 @@ def sensors_histories(self, filters, start_time_sec, end_time_sec, include_value
+            },
+            {
+                'activated': True,
+                'slug': '',
+                'name': 'm045',
+                'id': 3},
+            {..}]
+
+        """
+        url = self.sitemap['userlogs'] + '/tags'
+        response = yield self._http_client.fetch(url)
+        raise tornado.gen.Return(json.loads(response.body))
+
+    @tornado.gen.coroutine
+    def userlogs(self, start_time=None, end_time=None):
+        """
+        Return a list of userlogs in the database that has an start_time
+        and end_time combination that intersects with the given start_time
+        and end_time. For example of an userlog has a start_time before the
+        given start_time and an end time after the given end_time, the time
+        window of that userlog intersects with the time window of the given
+        start_time and end_time.
+
+        If an userlog has no end_time, an end_time of infinity is assumed.
+        For example, if the given end_time is after the userlog's start time,
+        there is an intersection of the two time windows.
+
+        Here are some visual representations of the time window intersections:
+
+                                Start       End
+        Userlog:                  [----------]
+        Search params:                 [-----------------]
+                                      Start              End
+
+                                              Start       End
+        Userlog:                                [----------]
+        Search params:              [-----------------]
+                                  Start              End
+
+                                 Start                End
+        Userlog:                  [--------------------]
+        Search params:                 [---------]
+                                     Start      End
+
+                                 Start     End
+        Userlog:                  [---------]
+        Search params:     [-------------------------]
+                         Start                      End
+
+                                              Start
+        Userlog:                                [-------------------*
+        Search params:              [-----------------]
+                                  Start              End
+
+                                                    End
+        Userlog:             *-----------------------]
+        Search params:                      [-----------------]
+                                          Start              End
+
+        Userlog:            *--------------------------------------*
+        Search params:              [-----------------]
+                                  Start              End
+        Parameters
+        ----------
+        start_time: str
+            A formatted datetime string used as the start of the time window
+            to query. Format: %Y-%m-%d %H:%m:%s.
+            Default: Today (local time) at %Y-%m-%d 00:00:00
```

Comment:
Definately say that time is in UTC, repeatedly in the doc strings and the examples.
Someone will email and ask "is time UTC or my local timezone"
Hope your using a strong and smart time parser on the server. Not the python built in.